eigen3 build is failing

Is anyone else seeing errors like this?
This just started happening a couple of weeks ago in our automated aicspylibczi builds. I assume Eigen updated something and the version being pulled by libCZI is not pinned.  Or else github-actions changed some compiler configuration.

```
/home/runner/work/aicspylibczi/aicspylibczi/cmake-build-debug/libCZI/Src/vendor/eigen3/src/eigen_ext/Eigen/src/Core/util/Macros.h:624:2: error: This compiler appears to be too old to be supported by Eigen
#error This compiler appears to be too old to be supported by Eigen
 ^
```

--- a/cmake/ExternalEIGEN3.cmake
+++ b/cmake/ExternalEIGEN3.cmake
@@ -4,7 +4,7 @@ ExternalProject_Add(
   eigen_ext
   PREFIX "vendor/eigen3"
   GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
-  #GIT_TAG        39142904cc2301628931481e8b331cc2d567e22f  # this is "known-to-work", just in case
+  GIT_TAG 3147391d946bb4b6c68edd901f2add6ac1f31f8c # this is "release version 3.4.0", we pin it to that version because as of today there are problems with the Linux-build
   CMAKE_ARGS
     -DEIGEN_BUILD_PKGCONFIG=OFF
     -DBUILD_TESTING=OFF

